Sofa/framework/Core/src/sofa/core/objectmodel/Data.h

```diff
@@ -111,6 +113,10 @@ class Data : public BaseData
     {
         m_value = ValueType(init.value);
         m_hasDefaultValue = true;
+        if constexpr (storeDefaultValue)
```

Comment:
By doing this aren't we introducing a parallel mechanism on the default value only for display purpose that doesn't have the same behavior as the actual mechanism of default value ? 

I mean, here, there is a possibility that the code itself defines a default value but it is never displayed because it doesn't satisfy this constexpr. Which may be missleading. Maybe doing it in a more violent way aka only allowing to have default value if we can store them would be preferable ?

Reply:
I am sorry but I don't understand everything in your comment.

Today, to the best of my knowledge, there is no mechanism to store the default value. This is what I add in this PR. So I don't understand when you say "parallel mechanism on the default value" and "actual mechanism of default value", because there are no such mechanism.

Reply:
The "actual mechanism of default value" refers to the actual default value mechanism that exists currently that just set the initial value of the data if not provided by the user in the scene file. Your mechanism will only store this value and display it to the user when storeDefaultValue is set to true, which may not be the case even-though a default value may exists.  

This new mechanism is of interest, my point was that this difference between the display and the fact that a default value exist can be missleading.

Reply:
If I understand correctly, you want to display the default value (if any), independently from the type of the Data?

Reply:
ok, got it